sway-core/src/semantic_analysis/namespace/arena.rs

```diff
@@ -281,6 +282,13 @@ impl NamespaceWrapper for NamespaceRef {
         }
     }
 
+    fn get_path_for_symbol(&self, symbol: &Ident) -> Vec<Ident> {
```

Comment:
since this uses the synonyms entry, maybe this could be named `get_canonical_path` or something similar?